pkg/vtpm/swtpm-vtpm/src/main.go

```diff
@@ -153,34 +229,114 @@ func main() {
 			continue
 		}
 
-		go handleRequest(conn)
+		id, err := bufio.NewReader(conn).ReadString('\n')
+		if err != nil {
+			log.Errorf("failed to read SWTPM ID from connection: %v", err)
+			continue
+		}
+
+		// Close the connection as soon as we read the ID,
+		// handle one request at a time.
+		conn.Close()
+
+		// Don't launch go routines, instead serve requests one by one to avoid
+		// using any locks, handle* functions access pids global variable.
+		err = handleRequest(id)
+		if err != nil {
+			log.Errorf("failed to handle request: %v", err)
+		}
 	}
 }
 
-func handleRequest(conn net.Conn) {
-	defer conn.Close()
+func handleRequest(id string) error {
+	request, id, err := processRequest(id)
+	if err != nil {
+		return fmt.Errorf("failed to process request: %w", err)
+	}
+
+	switch request {
+	// Domain manager is requesting to launch a new VM/App, run a new SWTPM
+	// instance with the given id.
+	case launchReq:
+		return handleLaunch(id)
+	// VM/App is being deleted, domain manager is sending a purge request,
+	// delete the SWTPM instance and clean up all the files.
+	case purgeReq:
+		return handlePurge(id)
+	// Domain manager is sending a delete request because it hit an error while
+	// starting the app (i.e. qemu crashed), so just remove SWTPM instance
+	// from the pids list and decrease the liveInstances count.
+	case deleteReq:
```

Comment:
Thanks for the comments here. They explain the use case. Nevertheless, I still do not understand why we don't have to remove the files in this case. They have already been created on the SWTMP launch, nope?

Reply:
I renamed it maybe now it makes more sense, but overall the reason for this request is to just preserve memory by asking the swtpm to stop running if it is no longer  needed, swtpm will clean up after itself when asked nicely, except the state directory, which we want to preserve because VM still exist.